### PR TITLE
Pod affinity/anti-affinity support Gt and Lt operators

### LINKED ISSUE #72506: Pod affinity/anti-affinity support Gt and Lt operators
**What would you like to be added**:
We know that node affinity/anti-affinity currently supports In, NotIn, Exists, DoesNotExist, Gt, Lt. But  pod affinity/anti-affinity only works with regular label selectors: In, NotIn, Exists, DoesNotExist

FYI:
https://bugzilla.redhat.com/show_bug.cgi?id=1462543;
https://github.com/kubernetes/website/issues/4137

**Why is this needed**:

_I don't know why Pod affinity/anti-affinity not support Gt and Lt operators. If anyone knows please tell me._

My initial requirement was to compare two labels (such as "20" and "30"). I am considering whether pod affinity/anti-affinity should support Gt and Lt operators?

/kind feature

### PR DESCRIPTION
**Pull request by wgliang**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #72506 

**Special notes for your reviewer**:

/kind feature

**Does this PR introduce a user-facing change?**:

```release-note
Pod affinity/anti-affinity support Gt and Lt operators.
```

### CODE REVIEW

**Comment by wgliang** [comment]
/assign @liggitt
@kubernetes/api-reviewers @kubernetes/sig-scheduling-pr-reviews

**Comment by k82cn** [comment]
will that make performance of pod affinity/anti-affinity worse?

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go [hunk 1]
```diff
@@ -46,6 +48,16 @@ func ValidateLabelSelectorRequirement(sr metav1.LabelSelectorRequirement, fldPat
 		if len(sr.Values) > 0 {
 			allErrs = append(allErrs, field.Forbidden(fldPath.Child("values"), "may not be specified when `operator` is 'Exists' or 'DoesNotExist'"))
 		}
+	case metav1.LabelSelectorOpGt, metav1.LabelSelectorOpLt:
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go]: This expands validation of all complex label selectors (replicasets, daemonsets, etc). Was that intended?

**Comment by liggitt** [comment]
Thanks for the pull request, but this needs to begin as a proposal discussing use cases and potential issues (like backwards compatibility, performance, and correctness implications), and be agreed on before implementation.

It should also clarify if the intent was only to expand selector syntax for affinity/anti-affinity, or for all complex label selectors (as this PR does).

/close

**Comment by liggitt** [comment]
Looking at the parent issue, it appears the intent was to target pod affinity/anti-affinity only. I'd still expect this change to first be proposed and agreed on. @kubernetes/sig-scheduling-proposals might know if there's an existing KEP for the affinity/anti-affinity feature that could be extended to cover this, and evaluated/agreed on.

**Comment by wgliang** [comment]
@liggitt  Thanks for your reply and comment, I will try to write a proposal for "Pod affinity/anti-affinity support Gt and Lt operators".

**Comment by wgliang** [comment]
@yastij @liggitt 
Although they are all related to affinity/anti-affinity, I don't think they are discussing one point.
#72479 is discussing how to use pod affinity/anti-affinity, but here is discussing extending pod affinity/anti-affinity.

**Comment by yastij** [comment]
My point was that there's some discussions around restricting affinity/anti-affinity, so I'd prefer having a KEP so we can include this into the discussions before moving to implementation.

**Comment by olehb** [comment]
This https://github.com/kubernetes/kubernetes/pull/77341 has been closed without merging. Are there still plans to implement this functionality?

**Pull request closed without merging**